src/engines/nwn2/itemproperty.cpp

```diff
@@ -76,10 +76,14 @@ uint8 ItemProperty::getItemPropertyCostTableValue() const {
 }
 
 bool ItemProperty::getIsItemPropertyValid() const {
+	// Check if marked invalid
+	if (_type == kItemPropertyInvalid || _type >= kItemPropertyMAX)
+		return false;
+
 	// Load the item properties row
 	const Aurora::TwoDAFile &twoDA = TwoDAReg.get2DA("itempropdef");
 	const size_t count = twoDA.getRowCount();
-	if (_type >= count)
+	if (_type >= (ItemPropertyType) count)
```

Comment:
Please do these casts the other way round, i.e. cast `_type` to a `size_t`. Probably won't hurt to use `static_cast` more too, instead of C-style casts.